app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,31 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @posts = Post.all
+    @user = User.find(params[:id])
+  end
 
   def show
-    @posts = Post.find(params[:id])
+    @post = Post.find(params[:id])
+  end
+end
+
+def new
+  @post = Post.new
+end
+
+def create
+  @post = current_user.posts.new(post_param[:id])
+  respond_to do |format|
+    format.html do
+      if @post.save
+        redirect_to user_post_path(current_user, @post), notice: 'Post was created.'
+      else
+        render :new, status: 'Error: Post was not created.'
+      end
+    end
   end
 end
+
+def post_param
+  params.require(:post).permit(:title, :text)
+end
```

Comment:
- Job well done in `config/route` file 🥇 
- 
![Capture](https://user-images.githubusercontent.com/63972670/200134853-7350535f-e4c4-4194-bd1d-ba04267f75ac.JPG)